### PR TITLE
Check size of keys when decoding

### PR DESCRIPTION
**Pull request by velzevur**
[Protect HTTP API from keys with wrong size](https://www.pivotaltracker.com/story/show/158383367)

### CODE REVIEW

File: apps/aecore/src/aec_base58c.erl [hunk 1]
```diff
@@ -107,6 +123,11 @@ pfx2type(<<"nm">>) -> name;
 pfx2type(<<"st">>) -> state;
 pfx2type(<<"pi">>) -> poi.
 
+type2size(poi) -> not_set;
+type2size(name) -> not_set;
+type2size(transaction) -> not_set;
+type2size(_) -> 32.
```
Comment (lucafavatella) [inline apps/aecore/src/aec_base58c.erl]: May I suggest explicitly listing each and every one of the types? This catch all may lead to surprising errors - especially if new types are introduced or size of any of the existing types changes.

Comment (lucafavatella) [inline apps/aecore/src/aec_base58c.erl]: Is size of type `signature` 32? I did not find that.

**Comment by velzevur** [comment]
Revisited key check and made it more assertive regarding new types.
Added integration tests for _shorter_ keys
Got a confirmation by @hanssv that signature size is 64 bytes.

**Comment by velzevur** [comment]
Rebased on top of master

**Comment by velzevur** [comment]
Rebased on top of NG changes

**Pull request merged**